Python tracing docs refresh (#13125)
getsentry/sentry-docs: docs/platforms/python/tracing/span-metrics/index.mdx

Span metrics allow you to extend the default metrics that are collected by tracing and track custom performance data and debugging information within your application's traces. There are two main approaches to instrumenting metrics:

1. [Adding metrics to existing spans](#adding-metrics-to-existing-spans)
2. [Creating dedicated spans with custom metrics](#creating-dedicated-metric-spans)

## Adding Metrics to Existing Spans

You can enhance existing spans with custom metrics by adding attributes. This is useful when you want to augment automatic instrumentation or add contextual data to spans you've already created.

```python
span = sentry_sdk.get_current_span()
if span:
    # Add individual metrics
    span.set_attribute("database.rows_affected", 42)
    span.set_attribute("cache.hit_rate", 0.85)
    
    # Add multiple metrics at once
    span.set_attributes({
        "memory.heap_used": 1024000,
        "queue.length": 15,
        "processing.duration_ms": 127
    })
```

### Best Practices for Span Attributes

When adding metrics as span attributes:

- Use consistent naming conventions (for example, `category.metric_name`)
- Keep attribute names concise but descriptive

> - Use appropriate data types (string, number, boolean, or (non-mixed) arrays of these types)

Reviewer: The double-nested parenthesis is a bit weird, would try to rewrite in a way that does not use this. For example:

```suggestion
- Use appropriate data types (string, number, boolean, or an array containing only one of these types)
```

## Creating Dedicated Metric Spans

> For more detailed operations, task, or process tracking, you can create custom dedicated spans that focus on specific metrics or attributes that you want to track. This approach provides better discoverability and more precise span configurations, however it can also appear to create more noise in your trace waterfall.

Reviewer: ```suggestion
For more detailed operations, tasks, or process tracking, you can create custom dedicated spans that focus on specific metrics or attributes that you want to track. This approach provides better discoverability and more precise span configurations, however it can also create more noise in your trace waterfall.
```

   - Use booleans for status flags
   - Use strings for categorical data
   - Use arrays when grouping related values

3. **Performance Considerations**
   - Consider the overhead of metric collection
   - Use sampling when collecting high-frequency metrics
   - Balance metric granularity with system performance

4. **Debugging and Monitoring**
   - Include correlation IDs for related operations
   - Add context that helps with troubleshooting

## Best Practices for Implementation

When implementing span metrics in your application:

1. **Start Small and Iterate**
   - Begin with basic metrics that directly relate to your debugging or performance monitoring needs
   - Add more detailed tracking as specific debugging needs emerge
   - Remove metrics that aren't providing actionable insights

2. **Maintain Consistency**
   - Use consistent naming patterns across your application
   - Document metric meanings and units in your codebase
   - Share common metrics across similar operations

3. **Focus on Actionability**
   - Track metrics that help diagnose specific issues
   - Consider what alerts or dashboard visualizations you'll want to create

>    - Ensure metrics can drive issue resolution or decision making

Author: I believe this was an ask from leadership to have more examples and "whys" of our functionality embedded in tracing documentation specifically.
